### PR TITLE
[parmetis] Fix build on x64-linux.

### PR DESCRIPTION
**Pull request by grospelliergilles**
- Change 'METIS' directory to 'metis' because the true name is 'metis'.
- Remove the non working (on linux) link 'METIS' to 'metis'.
- Use absolute path for GKLIB_PATH. Using relative path does not work
  for test HAVE_THREADLOCALSTORAGE in metis/GKlib/GKlibSystem.cmake

- #### What does your PR fix?  
  Fixes parmetis build on x64-linux

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <linux, windows, ...>, <No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/parmetis/portfile.cmake [hunk 1]
```diff
@@ -15,6 +15,7 @@ vcpkg_extract_source_archive_ex(
         fix-gklib-vs14-math.patch
         fix-root-cmakelist.patch
         fix-libparmetis-cmakelist.patch
+        fix-root-cmakelist-2.patch
         use_stdint.patch
```
Comment (NancyLi1013) [inline ports/parmetis/portfile.cmake:19]:
```suggestion
        use_stdint.patch
        fix-root-cmakelist-2.patch
```
Please place the new patch at the last.

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also update CONTROL file?

```
Version: 4.0.3
port-version: 4
```

**Comment by NancyLi1013** [comment]
Please also remove parmetis:x64-linux=fail from ci.baseline.txt.

https://github.com/microsoft/vcpkg/blob/807a7987612bbd702be9d9b8069124be62aff857/scripts/ci.baseline.txt#L1212

Could you please also help conform with the status on osx? @grospelliergilles

File: ports/parmetis/CONTROL [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Source: parmetis
-Version: 4.0.3-3
+Version: 4.0.3-4
```
Comment (NancyLi1013) [inline ports/parmetis/CONTROL:2]:
```suggestion
Version: 4.0.3
Port-Version: 4
```
Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#port-version

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/parmetis/CONTROL (hunk 2 above)
Comment (grospelliergilles) [reply ports/parmetis/CONTROL]: Thanks for the review. I have done the requested changes. I will try to check the status for 'osx' but I did not have a machine with this OS.

Comment (NancyLi1013) [reply ports/parmetis/CONTROL]: You can remove `parmetis:x64-linux=osx` from ci.baseline.txt to test the status on CI machine. If it still failed, you can try to fix the problem. If you cannot solve it, we can add it back.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your fix. @grospelliergilles

File: ports/parmetis/CONTROL (hunk 2 above)
Comment (grospelliergilles) [reply ports/parmetis/CONTROL]: I have removed the 'failed' status for 'osx' and it seems to work. The logs in the CI indicates that the build is OK.

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**